lib/teleterm/daemon/daemon_headless_watcher.go

```diff
@@ -103,14 +103,16 @@ func (s *Service) startHeadlessWatcher(cluster *clusters.Cluster) error {
 	ctx, cancel := context.WithCancel(s.closeContext)
 	s.headlessWatcherClosers[cluster.URI.String()] = cancel
 
+	log := s.cfg.Log.WithField("cluster", cluster.URI.String())
+
 	go func() {
 		defer cancel()
 		defer watcher.Close()
 
 		for {
 			select {
 			case <-ctx.Done():
-				s.cfg.Log.Debugf("Headless watcher for cluster %v closed with context error: %v", cluster.URI.String(), ctx.Err())
+				log.Debugf("Headless watcher closed with context error: %v", ctx.Err())
```

Comment:
Sometimes I think it's preferable to pass the error as part of the message rather than in `WithError` if you don't want to clog up the logs with full stack traces, but generally I'd agree.